dkpro-core-io-brat-asl/src/main/resources/org/dkpro/core/io/brat/bratTypes.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
I'd prefer not having a reader/format-specific type here. Types should be generally useful and be defined in one of the API modules - not in component or I/O modules.

Reply:
The test test__SingleDir__ThatDoesContainAnnFiles__AssumesEmptyAnnFiles() (which should be renamed to move the AssumesEmpty etc.. bit) illustrates why I needed to modify the structure of the test workspaces.

If I run BratReader with PARAM_SOURCE_LOCATION set to one of the resoures directories, the reader sees some files of the form *-ref.ann and goes looking for its corresponding *-ref.txt file. Not finding them, it raises an exception.

In order to test situations where I pass a directory path in PARAM_SOURCE_LOCATION, I therefore need to create an input directory that only contains .ann files for which there is a corresponding .txt file. That means erasing all the *-ref.ann files from the input directory. 

Obviously I don't want to erase them from src/resoources, so instead I copy those files over to a test workspace input dir, then erase the *-ref files from it.

Reply:
> I'd prefer not having a reader/format-specific type here. Types should be generally useful and be defined in one of the API modules - not in component or I/O modules.

I thought TypeMapping only applied to Brat files, hence the reason why I created a Brat-specific type for unmapped labels.

If the concept of TypeMapping is also used for other file formats, then I agree with you that we should have a generic, non-format-specific type that is used for un-mapped 'labels' seen in a file.

Reply:
Maybe it would be possible to use `PARAM_PATTERNS`: `"[-]*-ref.ann"`?

Reply:
That would defeat the point of the test. The point of the test is to ensure that if you have a sound brat directory ("sound" meaning that every .ann file has a .txt file) then you can run the BratReader on that directory without having to specify a pattern that identifies which files need to be analyzed.

Reply:
In that case i would suggest to simply set up multiple subfolders of src/test/resources each with exactly one scenario instead of assembling scenarios by copying data to these test workspaces - if necessary with an input and reference folder inside them.

Reply:
In that case i would suggest to simply set up multiple subfolders of src/test/resources each with exactly one scenario instead of assembling scenarios by copying data to these test workspaces - if necessary with an input and reference folder inside them.

Reply:
> In that case i would suggest to simply set up multiple subfolders of src/test/resources each with exactly one scenario instead of assembling scenarios by copying data to these test workspaces - if necessary with an input and reference folder inside them.

I think copying the original directory to a test input dir offers you the flexibility to "customize" a set of files for a particular test, without having to create a different directory of input files just to introduce a small test variant.

Also, I remembered that there was another reason for copying the files over. Basically, test__SingleDir__ThatDoesContainAnnFiles__AssumesEmptyAnnFiles() requires that you feed it a directory that contains .txt files, and no .ann files. But if this test uses the src/test/resources for PARAM_SOURCE_LOCATION, then it will end up creating empty .ann files in that directory. This means that the next time you run the same test, it will not be testing with the initial conditions that it's trying to test (because the .ann files are now present).